File: Perl_Python/functional/sort.pl
use strict;

$\ = "\n"; # some magic =P

my @arr = (0..9);

sub compare {
    return $a % 2 <=> $b % 2;
}


print join ', ', sort compare @arr;
print join ', ', sort {$a % 2 <=> $b % 2} @arr;

print join ', ', map { $_->[0] }
    sort { $a->[1] cmp $b->[1] }
    map { [$_, $_%2] }
    @arr;

File: Perl_Python/functional/sort.py
a = range(10)

def compare(x,y):
    return cmp(x%2, y%2)

print sorted(a, cmp=compare)
print sorted(a, cmp=lambda x,y: cmp(x%2, y%2))
print sorted(a, key=lambda x: x%2)
